xla/service/gpu/runtime3/sync_on_streams_thunk.cc

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2024 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "xla/service/gpu/runtime3/sync_on_streams_thunk.h"
+
+#include <string>
+#include <utility>
+
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "xla/service/gpu/thunk.h"
+#include "tsl/platform/errors.h"
+
+namespace xla {
+namespace gpu {
+
+absl::Status SyncOnStreamsThunk::ExecuteOnStream(const ExecuteParams& params) {
+  se::Stream* source_stream =
+      Thunk::GetStreamForExecution(source_stream_id_, params);
+  for (const auto& stream_id : wait_on_stream_ids_) {
```

Comment:
Single VLOG with absl::StrJoin will be better as for concurrent requests separate VLOGs hard to correlate